Pod/Classes/UI/NumberInputTextField.swift

```diff
@@ -171,9 +171,10 @@ public class NumberInputTextField: StylizedTextField {
         let numberPartiallyValid = type.checkCardNumberPartiallyValid(cardNumber) == .Valid
         
         if numberPartiallyValid {
-            let formatter = cardNumberFormatter
-            text = formatter.formattedCardNumber(unformattedCardNumber)
-            numberInputTextFieldDelegate?.numberInputTextFieldDidChangeText(self)
```

Comment:
This method is still called in `textField(self, shouldChangeCharactersInRange: _, replacementString: _)`:

```swift
        if !newValidation.contains(.NumberTooLong) {
            cardNumberFormatter.replaceRangeFormatted(range, inTextField: textField, withString: string)
            numberInputTextFieldDelegate?.numberInputTextFieldDidChangeText(self)
        } else if oldValidation == .Valid {
            // If the card number is already valid, should call numberInputTextFieldDidComplete on delegate
            // then set the text color back to normal and return
            numberInputTextFieldDelegate?.numberInputTextFieldDidComplete(self)
            super.textColor = _textColor
            return false
        } else {
            notifyNumberInvalidity()
        }
```

Instead of replacing the text though, the behavior is to imitate manual user input, which triggers all checks and behaviors (like adjusting the text color in the related issue) related to that.